src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MetricsEventSource.cs

```diff
@@ -0,0 +1,395 @@
+        private class CommandHandler
+        {
+            private AggregationManager? _aggregationManager;
+            private string _sessionId = "";
+
+            public void OnEventCommand(EventCommandEventArgs command)
+            {
+                try
+                {
+#if OS_SUPPORT_ATTRTIBUTES
+                    if (OperatingSystem.IsBrowser())
+                    {
+                        // AggregationManager uses a dedicated thread to avoid losing data for apps experiencing threadpool starvation
+                        // and browser doesn't support Thread.Start()
+                        //
+                        // This limitation shouldn't really matter because browser also doesn't support out-of-proc EventSource communication
+                        // which is the intended scenario for this EventSource. If it matters in the future AggregationManager can be
+                        // modified to have some other fallback path that works for browser.
+                        Logger.Error("", "System.Diagnostics.Metrics EventSource not supported on browser", "");
+                        return;
+                    }
+#endif
+                    if (command.Command == EventCommand.Update || command.Command == EventCommand.Disable ||
+                        command.Command == EventCommand.Enable)
+                    {
+                        if (_aggregationManager != null)
+                        {
+                            _aggregationManager.Dispose();
+                            _aggregationManager = null;
+                            Logger.Message($"Previous session with id {_sessionId} is stopped");
+                        }
+                        _sessionId = "";
+                    }
+                    if ((command.Command == EventCommand.Update || command.Command == EventCommand.Enable) &&
+                        command.Arguments != null)
+                    {
+                        if (command.Arguments!.TryGetValue("SessionId", out string? id))
+                        {
+                            _sessionId = id!;
+                            Logger.Message($"SessionId argument received: {_sessionId}");
+                        }
+                        else
+                        {
+                            _sessionId = System.Guid.NewGuid().ToString();
+                            Logger.Message($"New session started. SessionId auto-generated: {_sessionId}");
+                        }
+
+
+                        double defaultIntervalSecs = 1;
+                        Debug.Assert(AggregationManager.MinCollectionTimeSecs <= defaultIntervalSecs);
+                        double refreshIntervalSecs = defaultIntervalSecs;
+                        if (command.Arguments!.TryGetValue("RefreshInterval", out string? refreshInterval))
+                        {
+                            Logger.Message("RefreshInterval filter argument received: " + refreshInterval);
```

Comment:
For consistency with the other tracing:
```suggestion
                            Logger.Message($"RefreshInterval filter argument received: {refreshInterval}");
```